dask_ml/preprocessing/_encoders.py

```diff
@@ -154,12 +154,19 @@ def fit(
 
         return self
 
-    def _fit(self, X: Union[ArrayLike, DataFrameType], handle_unknown: str = "error"):
+    def _fit(
+        self,
+        X: Union[ArrayLike, DataFrameType],
+        handle_unknown: str = "error",
+        force_all_finite: bool = True,
+    ):
         X = check_array(
             X, accept_dask_dataframe=True, dtype=None, preserve_pandas_dataframe=True
         )
         if isinstance(X, np.ndarray):
-            return super(OneHotEncoder, self)._fit(X, handle_unknown=handle_unknown)
+            return super(OneHotEncoder, self)._fit(
+                X, handle_unknown=handle_unknown, force_all_finite=force_all_finite
+            )
```

Comment:
Should this be conditional on the version of scikit-learn? i.e. only pass it for SK_024?

```suggestion
            if SK_024:
                kwargs = dict(force_all_finite=force_all_finite)
            else:
                kwargs = {}
            return super(OneHotEncoder, self)._fit(
                X, handle_unknown=handle_unknown, **kwargs
            )
```